### PR TITLE
Model utils doc

### PR DESCRIPTION
**Pull request by sgugger**
This PR documents all the model utilities and fixes a few things in the `PreTrainedModel`/`TFPreTrainedModel`.

It introduces a new category I called "internal" that comes at the very end of the package documentation, which is aimed at documenting all our public internal methods (e.g., public in their modules, but not in the project init). This should be useful for people who tweak our code and reuse those pieces.

Preview of the new file in the docs is [here](https://64241-155220641-gh.circle-artifacts.com/0/docs/_build/html/internal/modeling_utils.html) (and you can navigate on the other pages if you want to check the smaller changes).

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -43,5 +43,5 @@ multi_line_output = 3
 use_parentheses = True
 
 [flake8]
-ignore = E203, E501, E741, W503
+ignore = E203, E501, E741, W503, W605
```
Comment (sgugger) [inline setup.cfg:46]: This is to be able to write LaTeX in docstrings. This category of errors is like "\s is not a valid escape character". I don't think it's too dangerous to ignore those compared to the benefit it adds, but I can revert and remove my formula if someone feels strongly against this.

**Review by jplu — APPROVED** [review]
Really nice!! LGTM.

File: setup.cfg (hunk 1 above)
Comment (sgugger) [reply setup.cfg:46]: Pinging @LysandreJik so you're aware of that change, merging now but we can revert later if you feel strongly against it.

Comment (LysandreJik) [reply setup.cfg:46]: I feel strongly about it, but in a good way :) It's a welcome addition!

**Pull request merged**